simple_network_sim/network_of_populations/__init__.py

```diff
@@ -49,18 +48,35 @@ class NetworkOfPopulation(NamedTuple):
     randomState: np.random.Generator
 
 
-def dateRange(startDate: dt.date, endDate: dt.date) -> Iterable[Tuple[dt.date, int]]:
+def dateRange(startDate: dt.date, endDate: dt.date) -> Iterable[dt.date]:
     """Generator of day and time from start date and end date
 
     :param startDate: Start date of the network
     :param endDate: End date of the network
-    :return: Generator of days as datetime.date and integer
+    :return: Generator of days as datetime.date
     """
     if startDate > endDate:
         raise ValueError("Model start date should be <= end date")
 
     for days in range(int((endDate - startDate).days)):
-        yield startDate + dt.timedelta(days=days + 1), days + 1
+        yield startDate + dt.timedelta(days=days + 1)
+
+
+def getInitialMultipliers(startDate: dt.date, multipliers: Dict[dt.date, loaders.Multiplier]):
+    """Queries the multipliers at the most recent date before
+    (and including) model start date. If no such date is found
+    returns neutral multipliers (1.0 for both).
+
+    :param startDate: Start date of the network
+    :param multipliers: dict of movement and contact multipliers
+    :return: Multipliers for initial day
+    """
+    dates = [d for d in sorted(list(multipliers.keys())) if d <= startDate]
+
+    if not dates:
+        return loaders.Multiplier(contact=1.0, movement=1.0)
+
+    return multipliers.get(dates[-1])
```

Comment:
Instead of sorting it we could do `max(dates)`.

Reply:
good point, fixed